pagopa/io-app
# feat(Bonus Pagamenti Digitali): [#174838425] Citizen enrollment and status

## Short description
Add the network and saga logic to get the citizen status and enroll the citizen to BPD

## List of changes proposed in this pull request
- add a [new endpoint](https://raw.githubusercontent.com/pagopa/io-services-metadata/master/bonus/specs/bpd/citizen.json) to generate citizen model based (a clone) on SIA spec api
- set backend spec to version 7.0.0 (bdp token in session)
- add `bdp` backend client
- start the `bdp` saga if feature flag is enabled
- change the `bdp` store shape: `onboarding` section has been removed, `enabled` section has been replaced in favor of citizen

## How to test
You can test it with the [io-dev-server](https://github.com/pagopa/io-dev-api-server). Make sure to run it with the last commit (branch `support-bpd`)
